lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -105,125 +105,49 @@ LinearLayout makeCgaLayout(CTALayoutAttr layout) {
   return ret.transposeOuts(outDimNames);
 }
 
-// Shrinks the output set of a layout function while leaving the input set
-// unchanged, by making high-order inputs in inDimName map to the same output.
-// Attempts to shrink down to desiredSize, but this is not always possible just
-// by modifying one the specified input dimension.
+// For each output dimension d, ensure that the layout's output size (i.e., its
+// codomain) does not exceed shape[d]. Do this without changing the size of the
+// layout's inputs (i.e., leave its domain unchanged).
 //
-// We do this by making the most-major inputs to the layout map to 0.  This
-// effectively duplicates data along that input dimension.  For example, this
-// layout has out-dim size 32:
+// This function is invariant to the order of the layout's input and output
+// dimensions.
 //
-//   L(register=1) = 8
-//   L(register=2) = 4
-//   L(register=4) = 1
-//   L(lane=1) = 2
-//   L(lane=2) = 16.
+// We achieve this by setting the largest value in each output dimension d to 0
+// because bases that map to a location larger the shape[d]
+// effectively duplicate along that dimension.  For example, consider a layout
+// with an output dimension size of 32, and we call ensureLayoutNotLargerThan to
+// shrink the output dimension size to 8:
```

Comment:
I notice some typos in this.  Here's what ChatGPT suggests, I think this is a good start.  https://chatgpt.com/share/e/0290a621-c78b-49e3-97cf-8a95b6b83030